bin/reth/src/config.rs

```diff
@@ -0,0 +1,80 @@
+//! Configuration files.
+use serde::{Deserialize, Serialize};
+
+/// Configuration for the reth node.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct Config {
+    /// Configuration for each stage in the pipeline.
+    // TODO(onbjerg): Can we make this easier to maintain when we add/remove stages?
+    pub stage: StageConfig,
+}
+
+/// Configuration for each stage in the pipeline.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct StageConfig {
+    /// Header stage configuration.
+    pub headers: HeadersConfig,
+    /// Body stage configuration.
+    pub bodies: BodiesConfig,
+    /// Sender recovery stage configuration.
+    pub senders: SendersConfig,
+}
+
+/// Header stage configuration.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct HeadersConfig {
+    /// The maximum number of headers to download before committing progress to the database.
+    pub commit_threshold: u64,
+    /// The maximum number of headers to request from a peer at a time.
+    pub downloader_batch_size: u64,
+    /// The number of times to retry downloading a set of headers.
+    pub downloader_retries: usize,
+}
+
+impl Default for HeadersConfig {
+    fn default() -> Self {
+        Self { commit_threshold: 10_000, downloader_batch_size: 1000, downloader_retries: 5 }
```

Comment:
should we set a higher retry count?

Reply:
This seems ok for now, at least I was able to make it to over 1.5m blocks. We should probably adjust all of these numbers at some point